tool/jt.rb

```diff
@@ -1050,85 +1051,98 @@ def test_compiler(*args)
   def test_cexts(*args)
     no_openssl = args.delete('--no-openssl')
     no_gems = args.delete('--no-gems')
+    test_to_run = args
 
-    # Test tools
+    begin
+      all_tests = %w(tools openssl minimum method module globals backtraces xopenssl gems)
+      tests = test_to_run.empty? ? all_tests : all_tests & args
+      tests.delete 'openssl' if no_openssl
+      tests.delete 'gems' if no_gems
+
+      tests.each do |test_name|
+        case test_name
+        when 'tools'
+          # Test tools
+          sh RbConfig.ruby, 'test/truffle/cexts/test-preprocess.rb'
+
+        when 'openssl'
+          # Test that we can compile and run some basic C code that uses openssl
+          if ENV['OPENSSL_PREFIX']
+            openssl_cflags = ['-I', "#{ENV['OPENSSL_PREFIX']}/include"]
+            openssl_lib = "#{ENV['OPENSSL_PREFIX']}/lib/libssl.#{SO}"
+          else
+            openssl_cflags = []
+            openssl_lib = "libssl.#{SO}"
+          end
 
-    sh RbConfig.ruby, 'test/truffle/cexts/test-preprocess.rb'
+          sh 'clang', '-c', '-emit-llvm', *openssl_cflags, 'test/truffle/cexts/xopenssl/main.c', '-o', 'test/truffle/cexts/xopenssl/main.bc'
+          out, _ = mx_sulong('lli', "-Dpolyglot.llvm.libraries=#{openssl_lib}", 'test/truffle/cexts/xopenssl/main.bc', capture: true)
+          raise out.inspect unless out == "5d41402abc4b2a76b9719d911017c592\n"
+
+        when 'minimum', 'method', 'module', 'globals', 'backtraces', 'xopenssl'
+          # Test that we can compile and run some very basic C extensions
+
+          gem_name = test_name
+          next if gem_name == 'xopenssl' && no_openssl
+          output_file = 'cext-output.txt'
+          dir = "#{TRUFFLERUBY_DIR}/test/truffle/cexts/#{gem_name}"
+          ext_dir = "#{dir}/ext/#{gem_name}/"
+          compile_cext gem_name, ext_dir, "#{dir}/lib/#{gem_name}/#{gem_name}.su"
+          case gem_name
+          when 'backtraces'
+            run_ruby "-I#{dir}/lib", "#{dir}/bin/#{gem_name}", err: output_file, continue_on_failure: true
+            actual = File.read(output_file).
+                gsub(TRUFFLERUBY_DIR, '').
+                gsub(/\/cext(_ruby)?\.rb:(\d+)/, '/cext\1.rb:n')
+            expected = File.read("#{dir}/expected.txt")
+            unless actual == expected
+              abort "C extension #{dir} didn't work as expected\nActual:\n#{actual}\nExpected:\n#{expected}\n"
```

Comment:
No need for `\n` at the end of `abort`, it's like `puts`. But it doesn't hurt either.